commons/src/main/java/io/aiven/kafka/tieredstorage/commons/transform/FetchChunkEnumeration.java

```diff
@@ -111,26 +108,26 @@ public InputStream nextElement() {
         final boolean isAtLastChunk = currentChunkId == lastChunkId;
         final boolean isSingleChunk = isAtFirstChunk && isAtLastChunk;
         if (isSingleChunk) {
-            final int toSkip = firstPosition - chunkStartPosition;
+            final int toSkip = range.from - chunkStartPosition;
             try {
                 chunkContent.skip(toSkip);
-                final int chunkSize = lastPosition - firstPosition;
-                chunkContent = new BoundedInputStream(chunkContent, chunkSize);
+                final int chunkSize = range.to - range.from;
+                chunkContent = new BoundedInputStream(chunkContent, chunkSize + 1);
             } catch (final IOException e) {
                 throw new RuntimeException(e);
             }
         } else {
             if (isAtFirstChunk) {
-                final int toSkip = firstPosition - chunkStartPosition;
+                final int toSkip = range.from - chunkStartPosition;
                 try {
                     chunkContent.skip(toSkip);
                 } catch (final IOException e) {
                     throw new RuntimeException(e);
                 }
             }
             if (isAtLastChunk) {
-                final int chunkSize = lastPosition - chunkStartPosition;
-                chunkContent = new BoundedInputStream(chunkContent, chunkSize);
+                final int chunkSize = range.to - chunkStartPosition;
+                chunkContent = new BoundedInputStream(chunkContent, chunkSize + 1);
```

Comment:
NIt: adding +1 during `chunkSize` calculation on the line above is a little easier to understand.